src/test/java/org/fogbowcloud/manager/core/ApplicationFacadeTest.java

```diff
@@ -12,1701 +12,2209 @@
+                .getFederationUser(Mockito.anyString());
 
-	@Test
-	public void testDeleteComputeOrderUnauthorizedOperation() throws Exception {
-		Mockito.doNothing().when(this.aaaController).authenticate(Mockito.anyString());
+        try {
+            // exercise
+            this.application.deleteCompute(order.getId(), FEDERATION_TOKEN_VALUE);
+            Assert.fail();
+        } catch (UnauthenticatedUserException e) {
+            // verify
+            Assert.assertEquals(OrderState.OPEN, order.getOrderState());
+        }
+    }
 
-		Order order = createComputeOrder();
-		OrderStateTransitioner.activateOrder(order);
+    // test case: When try calling the deleteCompute() method performing an operation without
+    // authorization, it must throw UnauthorizedRequestException and the Order remains in the same
+    // state.
+    @Test
+    public void testDeleteComputeOrderUnauthorizedOperation() throws Exception {
 
-		Mockito.doReturn(order.getFederationUser()).when(this.aaaController).getFederationUser(Mockito.anyString());
+        // set up
+        Order order = createComputeOrder();
+        OrderStateTransitioner.activateOrder(order);
 
-		Mockito.doThrow(new UnauthorizedRequestException()).when(this.aaaController).authorize(
-				Mockito.any(FederationUser.class), Mockito.any(Operation.class), Mockito.any(Order.class));
+        Mockito.doReturn(order.getFederationUser()).when(this.aaaController)
+                .getFederationUser(Mockito.anyString());
 
-		Mockito.doNothing().when(this.orderController).deleteOrder(Mockito.any(Order.class));
+        Mockito.doNothing().when(this.aaaController).authenticate(Mockito.anyString());
 
-		try {
-			this.application.deleteCompute(order.getId(), "");
-			Assert.fail();
-		} catch (UnauthorizedRequestException e) {
-			Assert.assertEquals(OrderState.OPEN, order.getOrderState());
-		}
-	}
+        Mockito.doThrow(new UnauthorizedRequestException()).when(this.aaaController).authorize(
+                Mockito.any(FederationUser.class), Mockito.any(Operation.class),
+                Mockito.any(Order.class));
 
-	@Test
-	public void testCreateComputeOrder() throws Exception {
-		ComputeOrder order = createComputeOrder();
+        Mockito.doNothing().when(this.orderController).deleteOrder(Mockito.any(Order.class));
 
-		Mockito.doNothing().when(this.aaaController).authenticate(Mockito.anyString());
+        try {
+            // exercise
+            this.application.deleteCompute(order.getId(), FEDERATION_TOKEN_VALUE);
+            Assert.fail();
+        } catch (UnauthorizedRequestException e) {
+            // verify
+            Assert.assertEquals(OrderState.OPEN, order.getOrderState());
+        }
+    }
 
-		Mockito.doReturn(order.getFederationUser()).when(this.aaaController).getFederationUser(Mockito.anyString());
+    // test case: When calling the createCompute() method the Order passed by parameter must return
```

Comment:
I did not understand the case. Are we trying to check the order state was correctly assigned after the exercise? If so, what was the state before the exercise?

Reply:
Yes, the previous 'OrderState' was 'null'. Anyway, I fixed the description of the test case.